src/mpid/ch4/shm/posix/posix_impl.h

```diff
@@ -168,4 +168,40 @@ extern MPIDI_POSIX_request_queue_t MPIDI_POSIX_sendq;
 extern MPIDI_POSIX_request_queue_t MPIDI_POSIX_recvq_unexpected;
 extern MPIDI_POSIX_request_queue_t MPIDI_POSIX_recvq_posted;
 
+/*
+ * Wrapper routines of process mutex for shared memory RMA.
+ * Called by both POSIX RMA and fallback AM handlers through CS hooks.
+ */
+#define MPIDI_POSIX_rma_mutex_init(mutex_ptr) do {                                  \
```

Comment:
My understanding is that, the macros define a set of operations that handle the mutex variable. Thus, it can be introduced together with the mutex definition in this commit, and be used by later commits. Could you please let me know if you have any better option ?

Reply:
Okay, I see what happened. I started from the top of the diff and wondered how `MPIDI_POSIX_rma_mutex_init` and `MPIDI_Devwin_t` are related. I missed `MPIDI_POSIX_win_t` in the bottom.

Now I think my true problem is that this patch mixes the definition of `MPIDI_POSIX_win_t` and the usage of it (in `MPIDI_Devwin_t`). How about having `MPIDI_POSIX_win_t` and these macro definitions in the first patch, and the modification to `MPIDI_Devwin_t` in the second patch?